modules/flowable-idm-engine/src/main/java/org/flowable/idm/engine/impl/IdmIdentityServiceImpl.java

```diff
@@ -212,4 +221,46 @@ public PrivilegeQuery createPrivilegeQuery() {
         return commandExecutor.execute(new GetUsersWithPrivilegeCmd(name));
     }
 
+    @Override
+    public PasswordEncoder getPasswordEncoder() {
+        return passwordEncoder;
+    }
+
+    @Override
+    public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public PasswordSalt getPasswordSalt() {
+        return passwordSalt;
+    }
+
+    @Override
+    public void setPasswordSalt(PasswordSalt passwordSalt) {
+        this.passwordSalt = passwordSalt;
+    }
+
+    private String getSalt() {
+        Object source = passwordSalt.getSource();
+        if (source instanceof String) return (String) source;
+        try {
+            Class<?> aClass = Class.forName("org.springframework.security.authentication.dao.SystemWideSaltSource");
```

Comment:
Do you see a possibility to implement this in a different way. I think the rest of the classes are defined in a clean way. But I would like to prevent this reflection code if possible.